guardrails/run/stream_runner.py

```diff
@@ -281,6 +250,32 @@ def step(
         iteration.outputs.validation_response = validated_fragment
         iteration.outputs.guarded_output = valid_op
 
+    def is_last_chunk(self, chunk: Any, api: Union[PromptCallableBase, None]) -> bool:
+        """Detect if chunk is final chunk"""
+        if isinstance(api, OpenAICallable):
+            if OPENAI_VERSION.startswith("0"):
+                finished = chunk["choices"][0]["finish_reason"]
+                return finished is not None
+            else:
+                finished = chunk.choices[0].finish_reason
+                return finished is not None
+        elif isinstance(api, OpenAIChatCallable):
+            if OPENAI_VERSION.startswith("0"):
+                finished = chunk["choices"][0]["finish_reason"]
+                return finished is not None
+            else:
+                finished = chunk.choices[0].finish_reason
+                return finished is not None
+        elif isinstance(api, LiteLLMCallable):
+            finished = chunk.choices[0].finish_reason
+            return finished is not None
+        else:
+            try:
+                finished = chunk.choices[0].finish_reason
+                return finished is not None
+            except (AttributeError, TypeError):
+                return False
```

Comment:
Is this also the approach for custom callables? I would propose we add a step where if the generator completes and finished is still false, we send a final empty chunk down the chain to the validators as well. Sort of a conservative catch all approach. That way people can continue to write their custom callables as just generators